### DISCUSSION TITLE
Dream IDE

### DISCUSSION THREAD

**Discussion by gabrielalexandrelopes** [zed-industries/zed#6570] [Feature Requests]
Hello everyone, 
This has been discussed extensively by you, but here it goes yet another IDE dream list:

1. Speed, simple, configurable UI (✔️ )
Zed is definitely in the right path, super responsive with a very clean interface. Would be nice to change the fonts of main application. FYI, Fleet from jet brains has a nice generic "tools" panels that can be placed on the left, right or bottom. Tools include the terminal, git, files, etc. Could be nice for Zed

2. Interaction with language servers (✔️ )
An essential element. Static checking, formatting, code navigation, etc. Plug-ins so that we can create our own languages 😛 

3. Integration of AI copilots (✔️ )
This is the future (present?)

4. Remote work (✔️ )
Very important for pair programming. Connect to social media tools? Streaming video of coding sessions? (The last 2 are not relevant for me)

5. interaction with Jupyter kernels to run code snippets directly in the editor.
The best implementation by far is hydrogen in Atom. This is the single reason while I keep using Atom/Pulsar at work. I run python, javascript, c++ and sometimes Mathematica. This could bring a lot of the AI/Jupyter users to Zed 

6. Git interaction (wip?)
Essential tool. My favourite implementation is Atom and git cola

7. File preview with real time updates.
This is very important for frontend developers. HTML, SVG, X3D, Markdown, LaTeX, etc

8. Build tools and Debugging
Important for all developers. Including flexible configuration options for building, running, deploying, cross-compiling in docker, etc. In practice, to have the ability to run scripts and interact with the output of debuggers

9. Plug-ins
All modern editors must have plugins 😛 

10. Built-in support for the most 10-15 used languages (wip?)
e.g. https://survey.stackoverflow.co/2023/#section-most-popular-technologies-programming-scripting-and-markup-languages)

Dear Zed team, do these align with your ideas?

**Comment by Tom-Boscher** [comment]
Hydrogen for Atom was definitely one the biggest main reason I used this IDE

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
